packages/wallets/src/features/cashier/flows/WalletWithdrawal/__tests__/WalletWithdrawal.spec.tsx

```diff
@@ -0,0 +1,141 @@
+
+describe('<WalletWithdrawal />', () => {
+    const originalWindowLocation = window.location;
+
+    beforeEach(() => {
+        Object.defineProperty(window, 'location', {
+            value: new URL('http://localhost/redirect?verification=1234'),
+            writable: true,
+        });
+    });
+
+    afterEach(() => {
+        Object.defineProperty(window, 'location', {
+            configurable: true,
+            value: originalWindowLocation,
+        });
+    });
+
+    it('should remove the `verification` param from the window url', () => {
+        const replaceStateSpy = jest.spyOn(window.history, 'replaceState');
+        mockUseActiveWalletAccount.mockReturnValue({
+            // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+            data: {
+                currency: 'USD',
+            },
+        });
+
+        // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+        mockUseCurrencyConfig.mockReturnValue({
+            getConfig: jest.fn(),
+            isSuccess: true,
+        });
+
+        render(<WalletWithdrawal />);
+
+        expect(replaceStateSpy).toBeCalledWith({}, '', 'http://localhost/redirect');
+    });
+
+    it('should render withdrawal email verification page if no verification code found', () => {
+        Object.defineProperty(window, 'location', {
+            value: new URL('http://localhost/redirect'),
+            writable: true,
+        });
+
+        mockUseActiveWalletAccount.mockReturnValue({
+            // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+            data: {
+                currency: 'USD',
+            },
+        });
+
+        // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+        mockUseCurrencyConfig.mockReturnValue({
+            getConfig: jest.fn(),
+            isSuccess: true,
+        });
+
+        render(<WalletWithdrawal />);
+        expect(screen.getByText('WithdrawalVerificationModule')).toBeInTheDocument();
+    });
+
+    it('should render withdrawal fiat module if withdrawal is for fiat wallet', () => {
+        mockUseActiveWalletAccount.mockReturnValue({
+            // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+            data: {
+                currency: 'USD',
+            },
+        });
+
+        mockUseCurrencyConfig.mockReturnValue({
+            // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+            getConfig: jest.fn(() => ({ is_fiat: true })),
+            isSuccess: true,
+        });
+
+        render(<WalletWithdrawal />);
+        expect(screen.getByText('WithdrawalFiatModule')).toBeInTheDocument();
+    });
+
+    it('should render withdrawal crypto module if withdrawal is for crypto wallet', () => {
```

Comment:
Could add a test case for the onClose function in withdrawal crypto module which removes the verification code as this part seems to not be covered yet
```
onClose={() => {
     setVerificationCode('');
}}
```